### PR TITLE
ENT-3758: RHEL 8.5 Cherry pick Requests

### PR DESCRIPTION
**Pull request by tlhmerry9**
Backported [62f4ee8](https://github.com/candlepin/subscription-manager/commit/911221395339566890e6b5d049d3d8158a5eaf3e) Warning message about the Depreciation of Syspurpose

Also Backported: [9a9b37c](https://github.com/candlepin/subscription-manager/commit/a36e10ff9afb50ae7fa2f586cac45435f42cf8a8)
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1928072
When SCA mode is used for selected organization, then do
not try to do auto-attach at all and print info message
We use the same message as is already used for
command attach --auto
Implemented this functionality, because old message was
generated by candlepin server. Look at following candlepin PR:
candlepin/candlepin#2947

### CODE REVIEW

**Review by ptoscano — CHANGES_REQUESTED** [review]
This is a backport of the PRs #2459 and #2421. Please backport their commits as such (`git cherry-pick -x ...`), so their history/changelog is preserved.

Note that cherry-picking commit 911221395339566890e6b5d049d3d8158a5eaf3e will need to not change also the files in `po/`, as they will be handled separately, either by us (keys.pot) or by Weblate (*.po).

**Comment by tlhmerry9** [comment]
? how did you find this commit?  I am looking and the only commit I see for PR #2421<https://github.com/candlepin/subscription-manager/pull/2421> is f087f2f<https://github.com/candlepin/subscription-manager/commit/f087f2f9187c7c10ed5d0d2a0462479a83bd1cf4>.  Thanks I appreciate it.

TIffany

________________________________

**Comment by tlhmerry9** [comment]
Will finish testing this on Thursday 4/15

File: syspurpose/src/syspurpose/cli.py [hunk 1]
```diff
@@ -348,6 +350,8 @@ def main():
         print(_("WARNING: Setting syspurpose in containers has no effect.\n"
               "Please run syspurpose on the host.\n"))
 
+    print(_("The 'syspurpose' command is deprecated and will be removed in a future major release."
+        " Please use the 'subscription-manager syspurpose' command going forward."), file=sys.stderr)
```
Comment (jirihnidek) [inline syspurpose/src/syspurpose/cli.py:354]: I cannot see this message in `./syspurpose/po/keys.pot`. Thus it cannot be translated for this branch.

**Review by jirihnidek — CHANGES_REQUESTED** [review]
I cannot see the all changes from 911221395339566890e6b5d049d3d8158a5eaf3e in this PR. Maybe some changes were part of another cherry-pick, but I would like to see warning message about deprecating of syspurpose in `keys.pot` file.

File: syspurpose/src/syspurpose/cli.py (hunk 1 above)
Comment (ptoscano) [reply syspurpose/src/syspurpose/cli.py:354]: We can update the translation catalogs afterwards, as separate change, like we did for other changes that affected the translatable messages. I will take care of it.

Comment (jirihnidek) [reply syspurpose/src/syspurpose/cli.py:354]: OK

**Review by jirihnidek — APPROVED** [review]
ACK

**Pull request merged**